### ISSUE TITLE
Next beta/feature release

### ISSUE THREAD

**Issue by Lineflyer** [cgeo/cgeo#9226]
AFAICS we do not have any pending PRs just now but already a smaller amount of new features already included on `master`.
As I prefer the approach of smaller cycles we should IMHO merge branches rather soon to push a new beta out.

We should check the labled "regressions" though:
https://github.com/cgeo/cgeo/issues?q=is%3Aopen+is%3Aissue+label%3ARegression

**Comment by fm-sys** [comment]
I like the idea of smaller release cycles, instead of too much extensive changes at ones and never-ending beta phases!

Looking at the [New backup handling - todo list (#9104)](https://github.com/cgeo/cgeo/issues/9104) these four todos are directly related to my latest changes:

- After a fresh install, if you try to create a backup, c:geo shows a dialog "database is empty, no backup needed" _(Btw. should we allow a settings backup when cgeo is newly installed at all? This message was introduced for secure reason, so that users do not overwrite their backup when actually trying to restore it...)_
- Change overwrite notice to something like `This will purge your oldest existing *backup dir* as of dd MMM, hh:mm - continue?`
- Simplify the "backup successful" message to something like `backup was created at <dir path>`
- add the backup state to `system information` (see #9049) 

Do you think any of them is critical and should definitely be included into the next release?

**Comment by moving-bits** [comment]
@fm-sys 
> Do you think any of them is critical and should definitely be included into the next release?

Just played around a bit with the backup/restore function on a fresh install in the emulator - works pretty well and is self-explaining. Thus I would be fine with having the extra changes in this area implemented in a later version.

**Comment by Lineflyer** [comment]
AFAICS those additions to backup have been merged as well now?!

Looking at the issues labeld as regression (link in initial posting), what about #9222 and #9225?
Both should be solvable I guess.

**Comment by moving-bits** [comment]
> AFAICS those additions to backup have been merged as well now?!

New backup/restore system haven been merged, the above mentioned four extras to it can be added later, IMHO.

> Looking at the issues labeld as regression (link in initial posting), what about #9222 and #9225?
> Both should be solvable I guess.

#9222 is something for @eddiemuc probably, and for #9225 see my comment https://github.com/cgeo/cgeo/issues/9225#issuecomment-712772466. Into what direction shall we continue there?

**Comment by eddiemuc** [comment]
> #9222 is something for @eddiemuc probably

I can take a look at it. Something for `release`?

**Comment by moving-bits** [comment]
If we're going to prepare a new beta soon, we could merge `master` to `release` and continue working on master.
So, yes, you would be on the safe side, if you prepare this directly for `release`.

**Comment by Lineflyer** [comment]
@fm-sys 
I found time today to work a bit on updating the user manual. Can you check if https://cgeo.droescher.eu/en/mainmenu/settings#backuprestore is correct?

**Comment by fm-sys** [comment]
> @fm-sys 
> I found time today to work a bit on updating the user manual. Can you check if https://cgeo.droescher.eu/en/mainmenu/settings#backuprestore is correct?

Looks good! Only some tables aren't below the pictures anymore but besides them. So if I open the webpage on my mobile the edited tables are completely hidden. Can you please fix this?

**Comment by moving-bits** [comment]
I've created PR #9260 which hides the old Mapsforge implementation without actually removing it. If we include this in the upcoming beta, this would be a step towards removing it completely, but there would still be an easy way to reactivate the map, if necessary.

Removing it completely (see #9246) could then be merged to master after branching.

**Comment by Lineflyer** [comment]
@moving-bits 
Regarding #9260 two questions:
1. What will happen if a user having V3 active upgrades to that version?
2. Shall we, instead of removing the whole setting instead disable it and just change the setting description to something like "This function has been disabled and will be removed in future versions.".

**Comment by Lineflyer** [comment]
> Looks good! Only some tables aren't below the pictures anymore but besides them. So if I open the webpage on my mobile the edited tables are completely hidden. Can you please fix this?

I put some smaller pictures leftbound intentionally. I know the mobile view is not really perfect. You are probably right to better provide the picture first and use full width for the table. Will change it next days.

**Comment by moving-bits** [comment]
> Regarding #9260 two questions:
> 1. What will happen if a user having V3 active upgrades to that version?

Since `Settings.useOldMapsforgeAPI()` will always return `false` after merging #9260, those users will work with `NewMap` automatically. No database migration or similar involved, no user action needed.
(Not sure, though, what happens if the user has activated an old theme, but I guess this should work as well, as the theme has to be paired up with a map file.)

> 2. Shall we, instead of removing the whole setting instead disable it and just change the setting description to something like "This function has been disabled and will be removed in future versions.".

Why should we do that? Whether it's disabled or removed, users cannot change its value any longer. Changing the text here would only lead to a new string to be translated, and as an place for announcing the planned removal I guess this is too hidden.

I would suggest to add a note to release notes instead, stating that "Old mapsforge map has been disabled and will be removed in future releases."

**Comment by bekuno** [comment]
3rd question: It possible that some user are using v3 maps. After upgrading they will have no offfline map anymore. That can be frustrating esp. without chance  of getting a new one. How we want to deal with this situation?

**Comment by moving-bits** [comment]
> 3rd question: It possible that some user are using v3 maps. After upgrading they will have no offfline map anymore. That can be frustrating esp. without chance of getting a new one. How we want to deal with this situation?

NewMap deals happily with version 3 map files, Mapsforge is downward-compatible.

**Comment by bekuno** [comment]
> NewMap deals happily with version 3 map files, Mapsforge is downward-compatible.

Oh yes, you are right. I forgot it.

**Comment by moving-bits** [comment]
@Lineflyer 
Regarding mf0.3: What do you prefer from support perspective?
- Alternative 1 (slowest)
  - show deprecation warning in upcoming beta 2020.10.x
  - hide mf0.3 (merge #9260) for the version after that (2020.11.x?)
  - remove mf0.3 (merge #9246) for the version after that (2020.12.x?)
- Alternative 2 (a bit faster)
  - show deprecation warning in upcoming beta 2020.10.x
  - remove mf0.3 (merge #9246) for 2020.11.x?
- Alternative 3 (fastest)
  - hide mf0.3 (merge #9260) in upcoming beta 2020.10.x
  - remove mf0.3 (merge #9246) in the version after that (2020.11.x?)

I would prefer option 2 or 3 because of not having to track things for such a long timespan, and I guess alternative 4 (remove mf0.3 / merge #9246 already in the upcoming beta) is probably a bit too fast :-)

**Comment by bekuno** [comment]
We have some fixes already in release. What about to make a new release (with mf0.3 deprecation) from it at first before start the next feature release?

**Comment by Lineflyer** [comment]
@fm-sys 
> Looks good! Only some tables aren't below the pictures anymore but besides them. So if I open the webpage on my mobile the edited tables are completely hidden. Can you please fix this?

Done!

@moving-bits 
> Why should we do that? Whether it's disabled or removed, users cannot change its value any longer. Changing the text here would only lead to a new string to be translated, and as an place for announcing the planned removal I guess this is too hidden.

> I would suggest to add a note to release notes instead, stating that "Old mapsforge map has been disabled and will be removed in future releases."

Agreed!

> I would prefer option 2 or 3 because of not having to track things for such a long timespan, and I guess alternative 4 (remove mf0.3 / merge #9246 already in the upcoming beta) is probably a bit too fast :-)

Option 3 seems totally fine for me.

**Comment by moving-bits** [comment]
> > I would suggest to add a note to release notes instead, stating that "Old mapsforge map has been disabled and will be removed in future releases."
> 
> Agreed!
> 
> > I would prefer option 2 or 3 because of not having to track things for such a long timespan, and I guess alternative 4 (remove mf0.3 / merge #9246 already in the upcoming beta) is probably a bit too fast :-)
> 
> Option 3 seems totally fine for me.

both done

**Comment by Lineflyer** [comment]
Now that the most urgent things are fixed with a new bugfix release we should find a good point in time (with respect to the status of `master`) to cut of a new beta/feature release by merging `master` to `release`.

Therefore we should focus merges to `master` on fixing regressions (refer to labeled issues) of existing and new features.

IMHO the current state is already quite stable and free of major regresions, however there is an older regression, where lot of users complain about: #8682 (Live map missing caches (OSM))
Any chance to spot a possible root cause and "try a fix" for beta?

**Comment by bekuno** [comment]
Maybe we need another bugfix release.
Please see https://github.com/cgeo/cgeo/issues/9150#issuecomment-721915489.
On the other side we have nearly 4 weeks until the end of the month, so this could be enougth time to have the feature release finished.

**Comment by Lineflyer** [comment]
> Maybe we need another bugfix release.
Please see #9150 (comment).

Thanks for the notice. Indeed we should fix this on `release` as probably the beta might not be productive until end of November.

**Comment by Lineflyer** [comment]
As soon as we have a fix for #9150 we should decide which open PRs need to be merged to `master` before starting into a beta phase.
@moving-bits  You are assigned to #9150, are you?

**Comment by bekuno** [comment]
The main/first part in #9150 is fixed. There is open a possible parsing problem for the cache page, see the latest state in https://github.com/cgeo/cgeo/issues/9150#issuecomment-722333614. We can currently only wait if GS changes the cache page.

**Comment by Lineflyer** [comment]
That sounds like an easy thing and will only happen end of November (if at all). So IMHO we should monitor this but it does not prevent a beta phase starting soon until we are satisfied with performance (and might include a change regarding #9150 on top in end of November).

Ready for merging `master`to `release` then?

**Comment by moving-bits** [comment]
> @moving-bits You are assigned to #9150, are you?

I'd implemented the original issue, now part 1, but I'm not working on other parts of it currently. It's probably easier to track if we close #9150 and open a separate issue, if they introduce other changes in this area (e. g. redirection).

**Comment by moving-bits** [comment]
> Ready for merging `master`to `release` then?

Generally yes from my side, haven't found much time the last weeks.
We may include #9291 (splash screen) if someone finds the time to cross-check, but including it is no must-have.

**Comment by bekuno** [comment]
I think that a solution for #9284 should be included in the next release.

**Comment by Lineflyer** [comment]
OK, solution for #9284 is done, #9291 is also included and a fix for #9354 

I will merge branches tonight after 8pm CET unless someone objects.

**Comment by Lineflyer** [comment]
For your attention:
- Branches have been merged (release is now equal to master)
- Beta version 2020.11.14-RC has been deployed on Google Play

**Please kindly select the appropriate branch for upcoming pull requests during the beta phase:**
- Fixes to existing features incl. regressions found during beta phase should target `release`
- New features, code refactoring and everything else should target `master`

**Comment by Lineflyer** [comment]
For information:
- Beta version seems to work smooth so far:
  - Not any relevant report on support mail
  - No high rate of crashes, except #9048 

I do see two ways forward:

1. Due to #9150 we should anyhow wait until beginning of December with releasing it. Maybe we find a fix for #9048 until then as well.
2. Release now, fix possible regressions we did not yet see once they are reported, in parallel fix #9048 and wait for #9150 and do another quick bugfix beginning of December.

Any preference?

**Comment by moving-bits** [comment]
If it's not too much extra work I would vote for releasing now to get the new things "out into the wild", giving us a bit more time to react if any problems should arise.
We still have the release branch to push any fixes regarding #9048 or #9150 (if needed).

So personal preference would be option 2, but I would be fine with option 1 as well.

**Comment by Lineflyer** [comment]
Thanks for feedback,
Then lets do option 2. More agile way ;)

I will prepare release:

- [ ] Check/Insert correct version name as headline in ```changelog-release.xml``` 
- [ ] Build release on CI server
- [ ] Perform smoke test with resulting ```cgeo-release.apk``` before continuing
- [ ] Upload and publish ```cgeo-release.apk``` to Google Play
- [ ] Tag the release on Github
- [ ] Publish the release on Github including upload of ```cgeo-release.apk``` to Github
- [ ] Publish release on F-Droid 
- [ ] Trigger notification to status.cgeo.org via CI server once the release is available on Google Play
- [ ] Post release info to Facebook 
- [ ] Post release info to Twitter

**Comment by moving-bits** [comment]
Version is "out in the wild", therefore closing this issue
